stable/prometheus-adapter/templates/hpa-external-metrics-cluster-role-binding.yaml

```diff
@@ -14,6 +14,6 @@ roleRef:
   name: {{ template "k8s-prometheus-adapter.name" . }}-external-metrics
 subjects:
 - kind: ServiceAccount
-  name: horizontal-pod-autoscaler
-  namespace: kube-system
+  name: {{ template "k8s-prometheus-adapter.serviceAccountName" . }}
```

Comment:
This should probably be undone. This is for the horizontal-pod-autoscaler serviceaccount in kube-system to be able to list the external metrics. Prometheus Adapter doesn't need to list/get/watch these metrics since he is the one creating them.

Reply:
Which begs the question how the `hpa-custom-metrics-cluster-role-binding.yaml` ever worked. The official prometheus adapter deployment [yaml](https://github.com/DirectXMan12/k8s-prometheus-adapter/blob/master/deploy/manifests/hpa-custom-metrics-cluster-role-binding.yaml) has it as kube-system:horizontal-pod-autoscaler.

`kubectl get clusterrole system:controller:horizontal-pod-autoscaler -o yaml` shows that HPA already has `custom.metrics.k8s.io` permission so it must come with HPA out of the box. Still, I think we should fix `hpa-custom-metrics-cluster-role-binding.yaml` as well.

Reply:
I removed the change from this PR for now and that can be done in a different one as these changes are not really related and require a separate explanation.